salt/modules/mssql.py

```diff
@@ -67,7 +71,71 @@ def _get_connection(**kwargs):
             connection_args[arg] = __salt__["config.option"](
                 "mssql." + arg, _DEFAULTS.get(arg, None)
             )
-    return pymssql.connect(**connection_args)
+    conn = pymssql.connect(**connection_args)
+    conn.autocommit(True)
+    return conn
+
+
+def _close_connection(cursor=None, connect=None):
+    """
+    Close the database connection and cursor connection
+    Args:
+        cursor (cursor object):
+        connect  (Connection object):
+
+    """
+    try:
+        if cursor:
+            cursor.close()
+    except pymssql.OperationalError:
+        pass
+    try:
+        if connect:
+            connect.close()
+    except pymssql.OperationalError:
+        pass
+
+
+def _to_rawstrings(text):
+    """
+    Escape an argument string to be suitable to the appropriate caller.
+    Special characters will be escaped.
+    For example, if a Windows path is passed in, special characters need to be escaped
+
+    Args:
+        text (str): a string argument to escape
+
+    Returns:
+        str: an escaped string
+    """
+    s = ""
+    if six.PY2:
+        if isinstance(text, six.string_types):
+            s = text.encode(encoding="string-escape")
+        elif isinstance(text, six.text_type):
+            s = text.encode(encoding="unicode-escape")
+    elif six.PY3:
+        if isinstance(text, six.text_type):
+            s = text.encode(encoding="unicode-escape")
+    return s
+
+
+def quote_identifier(identifier):
+    r"""
+    Return an identifier name (column, table, database, etc) escaped for MSSQL
+
+    This means surrounded by "[]" character and escaping this character inside.
+    It also means doubling the "'" character for pymssql internal usage.
+
+    :param identifier: the table, column or database identifier
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mssql.quote_identifier 'foo'bar'
```

Comment:
Here's a shell syntax error: `'foo'bar'` -- the single quote is not closed. Did you mean either `'foo.bar'` or `"foo'bar"` or something else?

Reply:
I'm trying to make the quote_identifier function private, and rename it to _quote_identifier, because it doesn't really make much sense to expose this function to the user

Reply:
This makes sense.

Reply:
This should be "foo'bar"